### ISSUE TITLE
pacman update hangs

### ISSUE THREAD

**Issue by integerbasic** [msys2/msys2#75]
waits forever in 'checking package integrity'. even with debug on, no apparent clues:

```
Packages (92) bash-completion-2.5-1  bsdcpio-3.3.1-1  bsdtar-3.3.1-1  coreutils-8.26-1  crypt-1.4-1  curl-7.54.0-1
              file-5.30-1  flex-2.6.4-1  gawk-4.1.4-2  gcc-libs-6.3.0-1  gettext-0.19.8.1-1  gmp-6.1.2-1
              gnupg-1.4.21-3  grep-3.0-1  icu-59.1-1  inetutils-1.9.4-1  info-6.3-1  libarchive-3.3.1-1
              libasprintf-0.19.8.1-1  libassuan-2.4.3-1  libcrypt-1.4-1  libcurl-7.54.0-1  libedit-3.1-20170329
              libexpat-2.2.0-2  libgettextpo-0.19.8.1-1  libgpg-error-1.27-1  libintl-0.19.8.1-1  liblzma-5.2.3-1
              liblzo2-2.10-1  libmetalink-0.1.3-1  libopenssl-1.0.2.k-1  libpcre-8.40-2  libpcre16-8.40-2
              libpcre32-8.40-2  libpcrecpp-8.40-2  libpcreposix-8.40-2  libreadline-7.0.003-1  libsqlite-3.19.3.0-1
              libtasn1-4.10-1  libxml2-2.9.2-3  m4-1.4.18-1  mingw-w64-x86_64-binutils-2.28-2
              mingw-w64-x86_64-bzip2-1.0.6-6  mingw-w64-x86_64-ca-certificates-20170211-2
              mingw-w64-x86_64-crt-git-5.0.0.4889.81b4f0e2-1  mingw-w64-x86_64-expat-2.2.3-1
              mingw-w64-x86_64-gcc-7.2.0-1  mingw-w64-x86_64-gcc-ada-7.2.0-1  mingw-w64-x86_64-gcc-fortran-7.2.0-1
              mingw-w64-x86_64-gcc-libgfortran-7.2.0-1  mingw-w64-x86_64-gcc-libs-7.2.0-1
              mingw-w64-x86_64-gcc-objc-7.2.0-1  mingw-w64-x86_64-gdb-8.0-2  mingw-w64-x86_64-gettext-0.19.8.1-2
              mingw-w64-x86_64-gmp-6.1.2-1  mingw-w64-x86_64-headers-git-5.0.0.4891.8629b425-1
              mingw-w64-x86_64-isl-0.18-1  mingw-w64-x86_64-libiconv-1.15-1
              mingw-w64-x86_64-libmangle-git-5.0.0.4760.d3089b5-1  mingw-w64-x86_64-libsystre-1.0.1-3
              mingw-w64-x86_64-libtasn1-4.12-1  mingw-w64-x86_64-libtre-git-r128.6fb7206-1
              mingw-w64-x86_64-libwinpthread-git-5.0.0.4850.d1662dc7-1  mingw-w64-x86_64-make-4.2.1-1
              mingw-w64-x86_64-mpfr-3.1.5.p10-1  mingw-w64-x86_64-ncurses-6.0.20170812-1
              mingw-w64-x86_64-openssl-1.0.2.l-1  mingw-w64-x86_64-p11-kit-0.23.7-1
              mingw-w64-x86_64-pkg-config-0.29.2-1  mingw-w64-x86_64-putty-0.69-1  mingw-w64-x86_64-python2-2.7.13-2
              mingw-w64-x86_64-python2-pip-9.0.1-3  mingw-w64-x86_64-python2-setuptools-36.2.7-1
              mingw-w64-x86_64-readline-7.0.003-1  mingw-w64-x86_64-tcl-8.6.7-1  mingw-w64-x86_64-tk-8.6.7-1
              mingw-w64-x86_64-tools-git-5.0.0.4849.1259532f-1  mingw-w64-x86_64-windows-default-manifest-6.4-3
              mingw-w64-x86_64-winpthreads-git-5.0.0.4850.d1662dc7-1  mingw-w64-x86_64-zlib-1.2.11-1  mpfr-3.1.5.1-3
              ncurses-6.0.20170708-2  openssl-1.0.2.k-1  pax-git-20161104.2-1  pcre-8.40-2  pkgfile-17-1  sed-4.4-2
              tzcode-2017.b-1  wget-1.19.1-1  winpty-0.4.3-1  xz-5.2.3-1  zlib-1.2.11-1

Total Installed Size:  841.50 MiB
Net Upgrade Size:       22.42 MiB

:: Proceed with installation? [Y/n]
debug: using cachedir: /var/cache/pacman/pkg/
debug: using cachedir: /var/cache/pacman/pkg/
checking keyring...
debug: looking up key 5F92EFC1A47D45A1 locally
debug: key lookup success, key exists
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
debug: key 5F92EFC1A47D45A1 found in cache
checking package integrity...
debug: found cached pkg: /var/cache/pacman/pkg/bash-completion-2.5-1-any.pkg.tar.xz
debug: sig data: iF4EABEIAAYFAlkuT64ACgkQX5LvwaR9RaGu+QD8DUigPAF6+uZFGydcPWv5U61ceX4bnAhc/l2jEooe7psA/1FiSLDcg2VlyGOI+6wiqO9wnDJj0auWlwg2srgw4Aeg
debug: checking signature for /var/cache/pacman/pkg/bash-completion-2.5-1-any.pkg.tar.xz
debug: 1 signatures returned
debug: fingerprint: AD351C50AE085775EB59333B5F92EFC1A47D45A1
debug: summary: valid
debug: summary: green
debug: status: Success
debug: timestamp: 1496207278
debug: exp_timestamp: 0
debug: validity: full; reason: Success
```

**Comment by elieux** [comment]
Some things to try:

1. Try installing just one package (and waiting some time). Maybe the process is just super slow?
2. Try checking [BLODA](https://cygwin.com/faq/faq.html#faq.using.bloda). You can use `pldd`, Process [Explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer)/[Monitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) or other tools to check for third-party DLLs injected into processes.
3. Check if `strace pacman -S ...` doesn't show something useful.

**Comment by integerbasic** [comment]
thank you elieux for your suggestions.  A few hours ago I took extreme measures and removed Msys2 altogether.  In the process, I found that a few files were in a very strange state; not owned by me, and I was unable to change the ownership (this is on a domain-managed system).  One of the problematic files was the pacman log file;  '/var/log/pacman' or something similar.  Another problematic file was one of the packages in /var/cache/pacman/pkg.  I was unable to delete these files manually, but after rebooting the system, they were gone.

After re-installation of Msys2, the problem no longer occurs.  So, if I see it again I will try some of your suggestions.  But I suspect pacman was probably being prevented from operating properly due to this file permission problem which is probably system-admin related, and has nothing to do with Msys2.

Thanks again.
